crates/oxc_linter/src/rules/typescript/no_misused_new.rs

```diff
@@ -113,6 +113,10 @@ impl Rule for NoMisusedNew {
             _ => {}
         }
     }
+
+    fn should_run(&self, ctx: &crate::rules::ContextHost) -> bool {
+        ctx.source_type().is_typescript()
+    }
 }
 
 #[test]
```

Comment:
on L146:


```
"class C { new(): C;}",
```

i think this should un on normal JS code?

Reply:
it defines that `: C` is returned.

from the rule docs:
> JavaScript classes may define a constructor method that runs when a class instance is newly created. TypeScript allows interfaces that describe a static class object to define a new() method (though this is rarely used in real world code). Developers new to JavaScript classes and/or TypeScript interfaces may sometimes confuse when to use constructor or new.

_https://typescript-eslint.io/rules/no-misused-new_

Tests without a declaration are passing in this case.

Reply:
you're right, it requires there's a return type. thanks for checking

https://github.com/typescript-eslint/typescript-eslint/blob/78fd2f4b530a3b703bb3be61b7cc825a24386c41/packages/eslint-plugin/src/rules/no-misused-new.ts#L102